src/common/weighted_shuffle.h

```diff
@@ -0,0 +1,29 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+// https://wandbox.org/permlink/SuTYLHHG8brNNieq
```

Comment:
much like https://godbolt.org, wandbox is more of a service of compilation. the link you are referencing here cannot be tracked by to the author. it reads

> created by
> anonymous
> 10 months ago

we cannot take the risk of using code from nowhere by nobody. but yeah, if the author transferred the ownership of copyright to you, so you can have the Signed-off-by, that's another story.

so i think the safe approach is to reference the stackoverflow link : https://stackoverflow.com/questions/50221136/c-weighted-stdshuffle/50223540 . as 

1. it is explicitly licensed under CC-AS. see https://stackoverflow.com/help/licensing
1. we can at least track to its original author "Bob__". but sadly, i have no idea if or how we can have his Signed-off-by.